src/test/java/com/scottlogic/AuthorPostSorterTest.java

```diff
@@ -0,0 +1,86 @@
+package com.scottlogic;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AuthorPostSorterTest {
+
+    @Test
+    void sort_withNull_returnsNull() {
```

Comment:
Nice naming style with the tests, in general instead of providing just `sort` you might want to do `authorPostSort` or `authorPostSorter` as when a full suite of tests is run you sometimes just get the failing test name and the file it fails in is slightly hidden. As you are doing lots of `sort` functions you might end up with lots of similar names so being able to easily differentiate them is useful.